CMakeLists.txt

```diff
@@ -373,3 +373,5 @@ if (OPM_ENABLE_PYTHON)
     add_definitions(-DEMBEDDED_PYTHON)
   endif()
 endif()
+
+target_link_libraries(opmcommon PUBLIC fmt::fmt)
```

Comment:
Looking at the [documentation](https://fmt.dev/latest/usage.html#building-the-library) this line could **arguably** be
```cmake
target_link_library(opmcommon PRIVATE fmt::fmt-header-only)
```
instead.  The above appears to link to a precompiled (`.a` or `.so`) version of the `fmt` library.

Reply:
Tried that - did not get it to work.

Reply:
> Tried that - did not get it to work.

In that case this (using `fmt`) may be more untenable.  If we do indeed need a precompiled third-party library then it is a much more expensive proposition.

Reply:
Well - that debian package (which I am using) has `libfmt.a` file which is linked to - but that is a compile time dependency - so I do not see a problem with that?

Reply:
it's a shared object on epel. anyways, see https://github.com/akva2/opm-common/commit/d8f0106ee3d5169215107fd28987b6bf146c6f6d (sorry you have to "distribute" across your commits).

tested on rhel and ubuntu.

the fmt-header-only target is fairly recent so we cannot rely on it (even ubuntu 20.04 is on fmt 6.1.2 which does not export it in the config files).

Reply:
Thank'll pick that up; but it does look like you have chosen a non optimal starting point for your fixes. I guess if you rebase on top of the tip of this branch I can just merge the changes right in?

Reply:
no, it sits on top of two of your branches and changes are across different commits. just make it happen, i don't care about authorship or whatever.

Reply:
OK - done; I have added `add_definitions(-DFMT_HEADER_ONLY)`. Thank you